javascript/apps/taiga/src/app/modules/project/story-detail/data-access/+state/reducers/story-detail.reducer.ts

```diff
@@ -77,6 +78,15 @@ export const reducer = createImmerReducer(
       return state;
     }
   ),
+  on(
+    KanbanApiActions.createStatusSuccess,
+    KanbanEventsActions.updateStatus,
+    (state, { status }): StoryDetailState => {
+      state.workflow?.statuses.push(status);
```

Comment:
Here we have to check if the status is created on the same workflow as the opened story detail. The event emits in the project channel but there we could have n workflows

Reply:
Not sure if understood well. I should compare the current workflow of the opened story detail with the workflow where the status is created, isn't it?

Something like this?

```ts
on(
    KanbanApiActions.createStatusSuccess,
    KanbanEventsActions.updateStatus,
    (state, { status, workflow }): StoryDetailState => {
      if (workflow === state.workflow?.slug) {
        state.workflow?.statuses.push(status);
      }

      return state;
    }
  ),
```